sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsModelFactory.cs

```diff
@@ -127,14 +192,41 @@ public static PartitionContext PartitionContext(string partitionId,
         /// <param name="offset">The offset of the event when it was received from the associated Event Hub partition.</param>
         /// <param name="enqueuedTime">The date and time, in UTC, of when the event was enqueued in the Event Hub partition.</param>
         ///
+        /// <remarks>
+        ///   This method is obsolete and should no longer be used.  Please use the overload with a string-based offset instead.
+        /// </remarks>
+        ///
+        [Obsolete(AttributeMessageText.LongOffsetOffsetParameterObsolete, false)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public static EventData EventData(BinaryData eventBody,
                                           IDictionary<string, object> properties = null,
                                           IReadOnlyDictionary<string, object> systemProperties = null,
                                           string partitionKey = null,
                                           long sequenceNumber = long.MinValue,
                                           long offset = long.MinValue,
                                           DateTimeOffset enqueuedTime = default) =>
-             new EventData(eventBody, properties, systemProperties, sequenceNumber, offset, enqueuedTime, partitionKey);
+            EventData(eventBody, properties, systemProperties, partitionKey, sequenceNumber, (offset > long.MinValue) ? offset.ToString(CultureInfo.InvariantCulture) : null, enqueuedTime);
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventHubs.EventData"/> class.
+        /// </summary>
+        ///
+        /// <param name="eventBody">The data to use as the body of the event.</param>
+        /// <param name="properties">The set of free-form event properties to send with the event.</param>
+        /// <param name="systemProperties">The set of system properties that accompany events read from the Event Hubs service.</param>
+        /// <param name="partitionKey">The partition hashing key associated with the event when it was published.</param>
+        /// <param name="sequenceNumber">The sequence number assigned to the event when it was enqueued in the associated Event Hub partition.</param>
+        /// <param name="offsetString">The offset of the event when it was received from the associated Event Hub partition.</param>
+        /// <param name="enqueuedTime">The date and time, in UTC, of when the event was enqueued in the Event Hub partition.</param>
+        ///
+        public static EventData EventData(BinaryData eventBody,
```

Comment:
This new override with default parameters, makes the following code not compile:
```csharpvar eventData = EventHubsModelFactory.EventData(new BinaryData(EncodeToUTF8(eventDataBody)), partitionKey: partitionKey);
```
Gives the error:
```
error CS0121: The call is ambiguous between the following methods or properties: 'EventHubsModelFactory.EventData(BinaryData, IDictionary<string, object>, IReadOnlyDictionary<string, object>, string, long, long, DateTimeOffset)' and 'EventHubsModelFactory.EventData(BinaryData, IDictionary<string, object>, IReadOnlyDictionary<string, object>, string, long, string, DateTimeOffset)'
```
Maybe remove the default parameter-values, so the conflict is fixed?